src/main/java/net/sf/jabref/gui/BasePanel.java

```diff
@@ -267,21 +262,26 @@ public ContentAutoCompleters getAutoCompleters() {
     public String getTabTitle() {
         StringBuilder title = new StringBuilder();
 
-        if (getBibDatabaseContext().getDatabaseFile() == null) {
-            title.append(GUIGlobals.UNTITLED_TITLE);
-
-            if (getDatabase().hasEntries()) {
-                // if the database is not empty and no file is assigned,
-                // the database came from an import and has to be treated somehow
-                // -> mark as changed
-                // This also happens internally at basepanel to ensure consistency line 224
-                title.append('*');
+        if (getBibDatabaseContext().getDatabase().getLocation() == DatabaseLocation.LOCAL) {
```

Comment:
To improve readability I would store that getDibDatabaseContext and getDatabase in a local variable